docs/design-docs/01-oci-support.md

```diff
@@ -0,0 +1,338 @@
+
+### Config chages
+
+The `Kptfile` structures for `upstream` and `upstreamLock` have `oci` in addition to `git` properties. The `type` property also has the string `oci` added as an accepted value.
+
+```yaml
+upstream:
+  type: oci
+  oci:
+    image: 'IMAGE:TAG'
+upstreamLock:
+  type: oci
+  oci:
+    image: 'IMAGE:DIGEST'
+```
+
+New verions of `kpt` will support existing `Kptfile`. The file structures and `git` functionality is unchanged.
+
+Existing versions of `kpt` will support `Kptfile` with `upstream` based on `git` for the same reason. The structure and meaning of existing fields is not changed.
+
+Existing versions of `kpt` will not support `Kptfile` with `upstream` based on `oci`. The `type` value, and missing `git` information will fail validation. The `kpt` binary used will need to be upgraded.
+
+### Command changes
+
+### `kpt pkg get`
+
+The argument that determines upstream today is parsed into `repo`, `ref`, and `path`, and is implicitly a `git` location.
+
+To support `oci`, it will be necessary to extract different values in a way that's unambiguous. Unfortunately, OCI image names have no Uri prefix, and are indistinguishable from a valid path or file name.
+
+To solve this, using [Helm](https://helm.sh/docs/topics/registries/#other-subcommands) as an example, the prefix `oci://` can be used. This ensures that selecting `oci` protocol isn't accidental, and it won't collide with other location formats that may be added.
+
+```shell
+# clone package as new folder
+kpt pkg get oci://us-docker.pkg.dev/the-project-id/the-repo-name/the-package:v3 my-package
+```
+
+Because OCI image reference already has a convention for `image:tag` references, using `:v3` should be used instead of `@v3` for version. It will be more intuitive how it relates to the registry, and easier to cut and paste values.
+
+### `kpt pkg get` sub-packages
+
+It is possible to use `kpt pkg get` to add sub-packages to a target location.
+
+Syntax for a sub-package target location is unchanged, it's a normal filesystem path.
+
+Syntax for an OCI sub-package source location requires the ability to tell when an image name ends and a sub-package path inside that image begins. In `git` this requires an explicit `.git` extension at the transition, and in `.oci` this requires double slash.
+
+```shell
+# clone sub-package as new sub-folder
+kpt pkg get oci://us-docker.pkg.dev/the-project-id/the-repo-name/the-package//simple/example:v3 my-package/simple/my-example
+```
+
+### `kpt pkg update`
+
+The command for update is not changed, but when the `upstream` is `oci` then the `@VERSION` is used to change the `upstream` image's `tag` or `digest` value.
+
+`kpt pkg update @v14` and `kpt pkg update DIR@v14` will assign `:v14` as the new upstream image tag.
+
+`kpt pkg update @sha256:xxxxx` and `kpt pkg update DIR@sha256:xxxxx` will assign `@sha256:xxxxx` as the new upstream image digest.
```

Comment:
The first case is like using a git branch or tag for the `@VERSION` that is resolved to a commit --- the second case is like using an exact git commit sha for the `@VERSION` that doesn't have any ambiguity. It matches can match exactly and only one thing, and that's what the `upstreamLock` will move onto (if it exists).

It could be stated altogether - it's just that the image tag and image digest syntax are a bit different from each other. I'll try adding add a yaml fragment as an example to see if that makes those line clearer

Reply:
I've added yaml fragments to clarify the results of the two cases